core/domain/exp_fetchers.py

```diff
@@ -412,3 +414,38 @@ def get_exploration_summaries_where_user_has_role(user_id):
         get_exploration_summary_from_model(exp_summary_model)
         for exp_summary_model in exp_summary_models
     ]
+
+
+def get_exploration_user_data(user_id, exp_id):
+    """Returns an ExplorationUserData domain object.
+
+    Args:
+        user_id: str. The Id of the user.
+        exp_id: str. The Id of the exploration.
+
+    Returns:
+        ExplorationUserData. The domain object corresponding to the given user
+        and exploration.
```

Comment:
Should this also explain the `None` that is sometimes returned?

Reply:
Updated the docstring now.